nstack-kotlin-android/src/test/java/dk/nodes/nstack/kotlin/managers/NetworkManagerTest.kt

```diff
@@ -0,0 +1,96 @@
+
+    private val guid = "guid"
+    private val version = "version"
+    private val oldVersion = "oldVersion"
+    private val device = "device"
+    private val osVersion = "osVersion"
+    private val lastUpdated = Date()
+    private val appOpenSettings = AppOpenSettings(
+            guid = guid,
+            version = version,
+            oldVersion = oldVersion,
+            lastUpdated = lastUpdated,
+            device = device,
+            osVersion = osVersion
+    )
+
+    @Before
+    fun setUp() {
+        Dispatchers.setMain(mainThreadSurrogate)
+    }
+
+    @Test
+    fun `Handle certificate expired`() {
+        val client = mockk<OkHttpClient>()
+        val manager = NetworkManager(client, "https://127.0.0.1", true)
+        every { client.newCall(any()).execute() } throws CertificateExpiredException()
+        every { client.newCall(any()).enqueue(any()) } throws CertificateExpiredException()
+
+        runManagerMethods(manager)
+    }
+
+    @Test
+    fun `Handle invalid certificate`() {
+        val client = mockk<OkHttpClient>()
+        val manager = NetworkManager(client, "https://127.0.0.1", true)
+        every { client.newCall(any()).execute() } throws CertPathValidatorException()
+        every { client.newCall(any()).enqueue(any()) } throws CertPathValidatorException()
+
+        runManagerMethods(manager)
+    }
+
+    @Test
+    fun `Handle IO Exception`() {
+        val client = mockk<OkHttpClient>()
+        val manager = NetworkManager(client, "https://127.0.0.1", true)
+        every { client.newCall(any()).execute() } throws IOException()
+        every { client.newCall(any()).enqueue(any()) } throws IOException()
+
+        runManagerMethods(manager)
+    }
+
+    private fun runManagerMethods(manager: NetworkManager) {
+        runBlockingTest { manager.loadTranslation("") }
+        runBlockingTest { manager.loadTranslation("", {}, {}) }
+
+        runBlockingTest { manager.fetchProposals({}, {}) }
+
+        runBlockingTest { manager.getCollection(1) }
+        runBlockingTest { manager.getCollectionItem(1, 1) }
+        runBlockingTest { manager.getLatestTerms(1, "da-DK", appOpenSettings) }
+        runBlockingTest { manager.getResponse("") }
+        runBlockingTest { manager.getRateReminder2(appOpenSettings) }
+
+        runBlockingTest { manager.postAppOpen(appOpenSettings, "da-DK") }
+        runBlockingTest { manager.postAppOpen(appOpenSettings, "da-DK", {}, {}) }
+        runBlockingTest { manager.postFeedback(appOpenSettings, "", "", "", null, FeedbackType.FEEDBACK) }
+        runBlockingTest { manager.postMessageSeen(appOpenSettings, 1) }
+        runBlockingTest { manager.postProposal(appOpenSettings, "", "", "", "", {}, {}) }
+        runBlockingTest { manager.postRateReminderAction(appOpenSettings, "") }
+        runBlockingTest { manager.postRateReminderAction(appOpenSettings, 1, "") }
+        runBlockingTest { manager.postRateReminderSeen(appOpenSettings, true) }
+    }
+
+}
```

Comment:
Changing it to `coVerify { ... }` does not work because manager is not mocked. The inverse `Assert.assertThrows` does not exist unless we bump JUnit to version 5.something and that didn't fly when I tried.